pyomo/contrib/gdpopt/ldsda.py

```diff
@@ -0,0 +1,558 @@
+
+        Parameters
+        ----------
+        external_var_value : list
+            The values of the external variables to be evaluated
+        search_type : str
+            The type of search, neighbor search or line search
+        config : ConfigBlock
+            GDPopt configuration block
+
+        Returns
+        -------
+        bool
+            weather the primal bound is improved
+        """
+        self.fix_disjunctions_with_external_var(external_var_value)
+        subproblem = self.working_model.clone()
+        TransformationFactory('core.logical_to_linear').apply_to(subproblem)
+
+        try:
+            with SuppressInfeasibleWarning():
+                try:
+                    TransformationFactory('gdp.bigm').apply_to(subproblem)
+                    fbbt(subproblem, integer_tol=config.integer_tolerance)
+                    TransformationFactory('contrib.detect_fixed_vars').apply_to(
+                        subproblem
+                    )
+                    TransformationFactory('contrib.propagate_fixed_vars').apply_to(
+                        subproblem
+                    )
+                    TransformationFactory(
+                        'contrib.deactivate_trivial_constraints'
+                    ).apply_to(subproblem, tmp=False, ignore_infeasible=False)
+                except InfeasibleConstraintException:
+                    return False, None
+                minlp_args = dict(config.minlp_solver_args)
+                if config.time_limit is not None and config.minlp_solver == 'gams':
+                    elapsed = get_main_elapsed_time(self.timing)
+                    remaining = max(config.time_limit - elapsed, 1)
+                    minlp_args['add_options'] = minlp_args.get('add_options', [])
+                    minlp_args['add_options'].append('option reslim=%s;' % remaining)
+                result = SolverFactory(config.minlp_solver).solve(
+                    subproblem, **minlp_args
+                )
+                # Retrieve the primal bound (objective value) from the subproblem
+                obj = next(subproblem.component_data_objects(Objective, active=True))
+                primal_bound = value(obj)
+                primal_improved = self._handle_subproblem_result(
+                    result, subproblem, external_var_value, config, search_type
+                )
+            return primal_improved, primal_bound
+        except RuntimeError as e:
+            config.logger.warning(
+                "Solver encountered RuntimeError. Treating as infeasible. "
+                "Msg: %s\n%s" % (str(e), traceback.format_exc())
+            )
+            return False, None
+
+    def _get_external_information(self, util_block, config):
+        """Function that obtains information from the model to perform the reformulation with external variables.
+
+        Parameters
+        ----------
+        util_block : Block
+            The GDPOPT utility block of the model.
```

Comment:
Consistency nit-pick
```suggestion
            The GDPopt utility block of the model.
```